csrc/mamba/mamba_ssm/selective_scan_fwd.cu

```diff
@@ -312,19 +312,20 @@ void selective_scan_fwd_launch(SSMParamsBase &params, cudaStream_t stream) {
     // kIsVariableB, kIsVariableC and kHasZ are all set to True to reduce binary size
     constexpr bool kIsVariableB = true;
     constexpr bool kIsVariableC = true;
-    constexpr bool kHasZ = true;
     BOOL_SWITCH(params.seqlen % (kNThreads * kNItems) == 0, kIsEvenLen, [&] {
-        BOOL_SWITCH(params.query_start_loc_ptr != nullptr , kVarlen, [&] {
-            using Ktraits = Selective_Scan_fwd_kernel_traits<kNThreads, kNItems, kNRows, kIsEvenLen, kIsVariableB, kIsVariableC, kHasZ,  kVarlen, input_t, weight_t>;
-            constexpr int kSmemSize = Ktraits::kSmemSize + kNRows * MAX_DSTATE * sizeof(typename Ktraits::scan_t);
-            dim3 grid(params.batch, params.dim / kNRows);
-            auto kernel = &selective_scan_fwd_kernel<Ktraits>;
-            if (kSmemSize >= 48 * 1024) {
-                C10_CUDA_CHECK(cudaFuncSetAttribute(
-                    (void *) kernel, cudaFuncAttributeMaxDynamicSharedMemorySize, kSmemSize));
-            }
-            kernel<<<grid, Ktraits::kNThreads, kSmemSize, stream>>>(params);
-            C10_CUDA_KERNEL_LAUNCH_CHECK();
+        BOOL_SWITCH(params.z_ptr != nullptr , kHasZ, [&] {
```

Comment:
Wasn't sure about this since IIRC we originally removed this due to wheel size reasons but looks reasonable to me:

```
Wheel dist/vllm-0.9.2rc2.dev95+g875e85bf6-cp38-abi3-linux_x86_64.whl is within the allowed size (368.15 MB).
```
(Arbitrarily comparing agasint #20707, I'm seeing 364.92 MB)




[2025-07-10T11:34:56Z] #32 0.752 Wheel dist/vllm-0.9.2rc2.dev124+g4a104607f-cp38-abi3-linux_x86_64.whl is within the allowed size (364.92 MB).
  |  

<br class="Apple-interchange-newline">


[2025-07-10T11:34:56Z] #32 0.752 Wheel dist/vllm-0.9.2rc2.dev124+g4a104607f-cp38-abi3-linux_x86_64.whl is within the allowed size (364.92 MB).
  |  

<br class="Apple-interchange-newline">